packages/block-library/src/query-total/edit.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	useBlockProps,
+	InspectorControls,
+	BlockControls,
+} from '@wordpress/block-editor';
+import {
+	PanelBody,
+	ToolbarGroup,
+	ToolbarDropdownMenu,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { resultsFound, displayingResults } from './icons';
+
+export default function QueryTotalEdit( {
+	attributes,
+	setAttributes,
+	className,
+} ) {
+	const { displayType } = attributes;
+
+	const [ totalResults, setTotalResults ] = useState( 0 );
+	const [ currentPage, setCurrentPage ] = useState( 1 );
+	const [ resultsPerPage, setResultsPerPage ] = useState( 10 );
+
+	// Fetch or calculate the total results, current page, and results per page.
+	useEffect( () => {
+		// Mock values for demonstration. Replace with actual query data.
+		const mockTotalResults = 12;
+		const mockCurrentPage = 1;
+		const mockResultsPerPage = 10;
```

Comment:
Does this mean this PR should still be a draft?

Reply:
@fabiankaegy actually these mock variable will only be displayed in the editor similar to the dummy results of the query loop.

Reply:
In that case we don't need this `useEfect` and the `useState` instead can just set those static variables directly 🤔

Reply:
Rectifying it right away🚀